### PR TITLE
Fix uncaught conversion error

### LINKED ISSUE #1089: Peloton crashes when inserting nonconvertible integer string
Peloton server crashes for the following session:

```
CREATE TABLE foo(id int);
INSERT INTO foo VALUES('h');
```

### PR DESCRIPTION
**Pull request by schedutron**
Fixes #1089.

### CODE REVIEW

File: src/include/type/value_factory.h [hunk 1]
```diff
@@ -1,597 +1,599 @@
...
+        case TypeId::VARCHAR: {
+          std::string str = value.ToString();
+          int32_t integer = 0;
+          try {
+            integer = stoi(str);
+          } catch (std::out_of_range &e) {
+            throw Exception(ExceptionType::OUT_OF_RANGE,
+                            "Numeric value out of range.");
+          } catch(std::invalid_argument &e){
+            throw Exception("Invalid input syntax for integer: \'" + str + "\'");
```
Comment (schedutron) [inline src/include/type/value_factory.h]: For some reason diff is improper for this commit. So I'm highlighting the lines modified.

Comment (schedutron) [inline src/include/type/value_factory.h]: For some reason diff is improper for this commit. So I'm highlighting the lines modified.

Comment (schedutron) [reply src/include/type/value_factory.h]: I've only added the lines 257, 258 and 259. Rest of the file is unchanged.

**Comment by schedutron** [comment]
@pervazea Can you please add the "ready for review" badge here?

**Comment by pervazea** [comment]
Please:
1. Fix the diff so we get proper diffs

**Comment by schedutron** [comment]
@pervazea Thanks, but why have you assigned me to my own PR?

**Comment by schedutron** [comment]
@pervazea Done!

**Comment by schedutron** [comment]
@pervazea Done! I've separated the changes into 2 commits. The 1st commit contains the code I added, and the 2nd commit contains the changes that the formatter wanted.

**Comment by schedutron** [comment]
I don't know why, but the formatting script was causing a build failure. For now, I'm removing that commit.

**Comment by schedutron** [comment]
@pervazea Cool. I'll start workng on this.

**Comment by schedutron** [comment]
@pervazea I can think of other types to be `BIGINT`, `SMALLINT`, `TINYINT` and `DECIMAL`. Would there be more?

**Comment by schedutron** [comment]
@pervazea I've added few tests. Should I also check the error string for each case?

**Comment by pervazea** [comment]
I can't think of any others at the moment.


________________________________

**Comment by schedutron** [comment]
Why did Jenkins fail? The Travis build is passing.

**Comment by schedutron** [comment]
@pervazea Are the tests good, or should I add exception message checks as well?

**Comment by schedutron** [comment]
@pervazea Any update here?

File: test/sql/insert_sql_test.cpp [hunk 2]
```diff
@@ -570,5 +570,42 @@ TEST_F(InsertSQLTests, UniqueColumn) {
...
+  std::string error_message;
+  std::unique_ptr<optimizer::AbstractOptimizer> optimizer(
+      new optimizer::Optimizer());
+
+  std::string create_table("CREATE TABLE foo (id1 int, id2 bigint,"
+                                         "id3 smallint, id4 tinyint,"
+                                         "id5 decimal);");
+  TestingSQLUtil::ExecuteSQLQuery(create_table);
+  // Insert an unconvertible int.
+  std::string query("INSERT INTO(id) foo VALUES('h');");
```
Comment (pervazea) [inline test/sql/insert_sql_test.cpp]: What is this insert trying to test? Is this what you intended?

Comment (schedutron) [reply test/sql/insert_sql_test.cpp]: It's trying to test whether an exception is raised when an invalid value is inserted.

**Comment by schedutron** [comment]
@pervazea I've fixed merge conflicts that were present.

**Comment by schedutron** [comment]
@apavlo I've added the tests, can you please remove the `needs-tests` label and add back the `in-progress` label?

**Comment by schedutron** [comment]
I don't know why the Travis build is failing. The few last lines of the build log are:
```
[100%] Built target coveralls

travis_time:end:149f4a80:start=1519668488054710457,finish=1519668523389125663,duration=35334415206
[0K
[32;1mThe command "make coveralls" exited with 0.[0m

Done. Your build exited with 1.
```
Why is it exiting with 1?

**Comment by pervazea** [comment]
You need to go back in the logs to see the cause of failure.

Select the failing Travis build. You'll see on the right, a button for "raw log", select that.

Now go back from the end, through the log, and find which step of the build failed. Often, it will be one of the tests.


When I do that, I see it is the insert_sql_test. I pasted the portion of the output below, for you.


70: 2018-02-26 18:03:48 [test/sql/testing_sql_util.cpp:131:ExecuteSQLQueryWithOptimizer] INFO  - Statement executed. Result: SUCCESS
70: [0;32m[       OK ] [mInsertSQLTests.InsertIntoSelectColumn (247 ms)
70: [0;32m[ RUN      ] [mInsertSQLTests.UniqueColumn
70: 2018-02-26 18:03:48 [src/catalog/catalog.cpp:282:CreateTable] DEBUG - Added a UNIQUE index on st in t.
70: 2018-02-26 18:03:48 [test/sql/testing_sql_util.cpp:58:ExecuteSQLQuery] INFO  - Query: select st from t;
70: 2018-02-26 18:03:48 [src/codegen/operator/table_scan_translator.cpp:40:TableScanTranslator] DEBUG - Constructing TableScanTranslator ...
70: 2018-02-26 18:03:48 [src/codegen/operator/table_scan_translator.cpp:53:TableScanTranslator] DEBUG - Finished constructing TableScanTranslator ...
70: 2018-02-26 18:03:48 [test/sql/testing_sql_util.cpp:93:ExecuteSQLQuery] INFO  - Statement executed. Result: SUCCESS
70: 2018-02-26 18:03:48 [test/sql/testing_sql_util.cpp:58:ExecuteSQLQuery] INFO  - Query: select st from t;
70: 2018-02-26 18:03:48 [test/sql/testing_sql_util.cpp:93:ExecuteSQLQuery] INFO  - Statement executed. Result: SUCCESS
70: 2018-02-26 18:03:48 [test/sql/testing_sql_util.cpp:58:ExecuteSQLQuery] INFO  - Query: select st from t;
70: 2018-02-26 18:03:48 [test/sql/testing_sql_util.cpp:93:ExecuteSQLQuery] INFO  - Statement executed. Result: SUCCESS
70: [0;32m[       OK ] [mInsertSQLTests.UniqueColumn (118 ms)
70: [0;32m[ RUN      ] [mInsertSQLTests.BadTypes
70: [0;32m[       OK ] [mInsertSQLTests.BadTypes (42 ms)
70: [0;32m[ RUN      ] [mInsertSQLTests.NonExistentTable
70: unknown file: Failure
70: C++ exception with description "Database default_database already exists" thrown in the test body.
70: [0;31m[  FAILED  ] [mInsertSQLTests.NonExistentTable (1 ms)
70: [0;32m[----------] [m10 tests from InsertSQLTests (1494 ms total)
70:
70: [0;32m[----------] [mGlobal test environment tear-down
70: [0;32m[==========] [m10 tests from 1 test case ran. (1494 ms total)
70: [0;32m[  PASSED  ] [m9 tests.
70: [0;31m[  FAILED  ] [m1 test, listed below:
70: [0;31m[  FAILED  ] [mInsertSQLTests.NonExistentTable


________________________________

**Comment by schedutron** [comment]
@pervazea Thanks a lot, it was much difficult for me to spot the failure in the raw log, without syntax highlighting. I'll fix the tests soon.

**Comment by schedutron** [comment]
I was missing the database deletion after the tests. After adding it back, I ran the tests locally and they passed. Hopefully, the Travis build should pass this time.

**Comment by schedutron** [comment]
@pervazea Thanks, the Travis build is passing now.

**Comment by schedutron** [comment]
@pervazea Thanks, the Travis build is passing now.

**Comment by schedutron** [comment]
@pervazea Any update here?

**Comment by schedutron** [comment]
@apavlo Can you please assign someone to review this PR, it's been idle for a while. I need feedback to work further :)

**Comment by schedutron** [comment]
@pervazea Sorry to bother you, but again, any update here? The review won't take much time :)

**Comment by pervazea** [comment]
The code itself looked ok. A couple of issues though:
1. Why remove the copyright header on value_factory.h?
2. We need to fix the line endings. If you change your line endings to Unix format and re-submit value_factory.h, we should get proper diffs on Github.

Do you do your editing on a Windows machine? The correct fix for the line ending issue will be to configure the Peloton repository and git to automatically handle the line ending conversions. I'm examining doing that. However, the change will need to propagate into user's forked repos, before it provides benefit.

**Comment by schedutron** [comment]
@pervazea I'm working on a Mac. I don't know how the copyright header got removed, will add it back in the next commit. And I'll also try to fix the line endings and see if the diffs get proper.

Also, you can ask contributors to update their branches with master after you have resolved this diff issue.

**Comment by schedutron** [comment]
@pervazea `git diff --ignore-space-at-eol` is showing the correct diff.

**Comment by schedutron** [comment]
@pervazea I ran the source through a Python script and removed the trailing whitespaces. The diff is now displaying fine.

**Comment by schedutron** [comment]
@pervazea If all seems good now, can I run the formatter script on the source? I committed with the `--no-verify` flag so that you can first review the changes I added.

**Comment by schedutron** [comment]
@pervazea Thanks for the entire guidance :)

**Pull request merged**